rules/no-await-in-loop.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * @fileoverview Rule to disallow uses of await inside of loops.
+ * @author Nat Mote
+ */
+"use strict";
+
+// Node types which are considered loops.
+var loopTypes = new Set([
+  'ForStatement',
+  'ForOfStatement',
+  'ForInStatement',
+  'WhileStatement',
+  'DoWhileStatement',
+]);
+
+// Node types at which we should stop looking for loops. For example, it is fine to declare an async
+// function within a loop, and use await inside of that.
+var boundaryTypes = new Set([
+  'FunctionDeclaration',
+  'FunctionExpression',
+  'ArrowFunctionExpression',
+]);
+
+module.exports = function(context) {
+  return {
+    // babel-eslint transpiles AwaitExpressions to YieldExpressions, but the actual node kind is
+    // still available in _babelType.
+    YieldExpression: function(node) {
+      if (node._babelType === 'AwaitExpression') {
```

Comment:
Yeah I wonder if we should still transpile it to a yield (most of those transforms are only done since it causes other eslint rules to error). But yeah `_babelType` at least helps here!